platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceBean.java

```diff
@@ -525,6 +536,37 @@ protected BundleContext getContext() {
     }
   }
 
+  @Override
+  public void restart() {
```

Comment:
Not sure I want to do this right now. There are differences between the two in terms of how we log or report and I don't want to complicate this even more by parameterizing both. Even if I do this right now, it still bleeds the internal implementation out to both bundles as they still need to know they require access to read/write the file and to the properties. This is a perfect case where we should have a standalone utility library that is defined as a bundle and not and embeddable library and which should extend its privileges and define a new restart permission as opposed to bleeding implementation details out.

Reply:
How difficult would it be to create that new "restart" service? Could we work that now in a separate PR?

Reply:
Definitely not in this PR. I don't think it would be too complicated. But here, I kinda need it to report every step of the way since migration:export actually generates entries in a report as opposed to logging. This makes it really tightly coupled anyway.

Reply:
I think we should seriously consider the separate PR to address the code duplication and coupling with how the system is restarted (system properties, wrapper, etc.), and also to avoid the technical debt.